packages/react-menu/src/components/Menu/useMenu.tsx

```diff
@@ -60,5 +72,54 @@ export const useMenu = (props: MenuProps, ref: React.Ref<HTMLElement>, defaultPr
     [setOpen],
   );
 
+  useMenuPopup(state);
+  useOnClickOutside({ element: document, refs: [state.menuPopupRef, triggerRef], callback: () => setOpen(false) });
+  return state;
+};
+
+const useMenuPopup = (state: MenuState) => {
+  // TODO use Popper for the popup ref
+  const menuPopupRef = React.useRef<HTMLElement>(null) as React.MutableRefObject<HTMLElement>;
+  state.menuPopupRef = menuPopupRef;
+  state.menuPopup.children = (Component, originalProps) => {
+    const newProps = { 'aria-labelledby': state.triggerId, ...originalProps };
+
+    if (state.onHover && !state.onContext) {
+      newProps.onMouseEnter = (e: React.MouseEvent<HTMLElement>) => {
+        state.setOpen(true);
+        originalProps?.onMouseEnter?.(e);
+      };
+      newProps.onMouseLeave = (e: React.MouseEvent<HTMLElement>) => {
+        state.setOpen(false);
+        originalProps?.onMouseLeave?.(e);
+      };
+    }
+
+    newProps.onBlur = (e: React.FocusEvent<HTMLElement>) => {
+      state.setOpen(false);
+      originalProps?.onBlur?.(e);
+    };
+
+    newProps.onKeyDown = (e: React.KeyboardEvent<HTMLElement>) => {
+      originalProps?.onKeyDown?.(e);
+      const keyCode = getCode(e);
+
+      if (keyCode !== keyboardKey.Escape) {
+        return;
+      }
+
+      state.setOpen(false);
+    };
+
+    return (
+      // Shorthand render funct types have issues
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      <Component {...newProps} ref={state.menuPopupRef}>
+        {state.menuList}
+      </Component>
+    );
+  };
```

Comment:
```suggestion
 return React.createElement(
      Component as React.ElementType,
      {
        ...newProps,
        ref: state.menuPopupRef,
      },
      state.menuList,
    );
```

nit: I propose to replace with `.createElement()` call + type cast, at least we will not ignore everything

Reply:
was stuck with that for a while before I gave up and ignored... thanks for the protip